src/safeposix/syscalls/net_calls.rs

```diff
@@ -1399,23 +1641,52 @@ impl Cage {
                                     }
                                     // get the socket pipe, write to it, and return bytes written
                                     let sockinfo = &sockhandle.unix_info.as_ref().unwrap();
+                                    //When the message does not fit into the send buffer of the
+                                    // socket, send() normally
+                                    // blocks, unless the socket has been placed in
+                                    // nonblocking I/O mode.  In nonblocking mode it would fail with
+                                    // the error EAGAIN or
+                                    // EWOULDBLOCK in this case.
                                     let mut nonblocking = false;
                                     if sockfdobj.flags & O_NONBLOCK != 0 {
                                         nonblocking = true;
                                     }
                                     let retval = match sockinfo.sendpipe.as_ref() {
+                                        //sendpipe is available in unix socket info
+                                        //it is needed to send the message found in buf
+                                        //to the connected socket
                                         Some(sendpipe) => {
                                             sendpipe.write_to_pipe(buf, buflen, nonblocking) as i32
                                         }
+                                        //sendpipe is not available in unix socket info
+                                        //transmission of data is not possible so return with error
                                         None => {
-                                            return syscall_error(Errno::EAGAIN, "writev", "there is no data available right now, try again later");
+                                            panic!("sendpipe is not available in unix socket info in send syscall");
```

Comment:
Weird to change this.  Why panic?

Reply:
Looking at it again, I suppose it is possible that send is called on a unix socket that isn't connected, so we wouldn't want to panic. Changing to an error, thanks!